src/include/compact_set.h

```diff
@@ -147,20 +146,23 @@ class compact_set_base {
     if (set) {
       assert(this == p.set);
       set->erase(p.it);
-      if (set->empty())
+      if (set && set->empty()) {
 	free_internal();
+      }
     }
   }
   size_t erase (const T& t) {
     if (!set)
       return 0;
     size_t r = set->erase(t);
-    if (set->empty())
+    if (set && set->empty()) {
 	free_internal();
+    }
     return r;
   }
   void clear() {
-    free_internal();
+    if (set)
+      free_internal();
```

Comment:
what's the point of this change when you also had to change free_internal() to assert(set)?